pkg/forwarder/transactions_file_storage.go

```diff
@@ -56,6 +56,11 @@ func newTransactionsFileStorage(
 // Serialize serializes transactions to the file system.
 func (s *transactionsFileStorage) Serialize(transactions []Transaction) error {
 	s.telemetry.addSerializeCount()
+
+	// Reset the serializer in case some transactions were serialized
+	// but `GetBytesAndReset` was not called because of an error.
+	_, _ = s.serializer.GetBytesAndReset()
```

Comment:
Shouldn't this be handled when we have an error ? What if multiple goroutine are calling the serializer ?

Reply:
I think `transactionsFileStorage.Serialize` should put the serializer in a known state before using it. This is what this line does. 

`transactionContainer` is thread safe and so `transactionsFileStorage` do not need to be thread safe as it is called only by `transactionContainer`.